src/openms/source/CHEMISTRY/EnzymaticDigestion.cpp

```diff
@@ -202,34 +227,14 @@ namespace OpenMS
         }
         return score_missed - score_cleave > log_model_threshold_;
       }
-      else // naive digestion
-      {
-        // R or K at the end and not P afterwards
-        return (*iterator == 'R' || *iterator == 'K') &&
-               ((iterator + 1) == protein.end() || *(iterator + 1) != 'P');
-      }
-      break;
-    case ENZYME_TRYPSIN_P:
-      if (use_log_model_)
-      {
-        throw Exception::InvalidParameter(__FILE__, __LINE__, __PRETTY_FUNCTION__, String("EnzymaticDigestion: enzyme '") + NamesOfEnzymes[ENZYME_TRYPSIN_P] + " does not support logModel!");
-      }
-      else
-      {
-        // R or K at the end,  presence of P does not matter
-        return (*iterator == 'R' || *iterator == 'K');
-      }
-      break;
-    default:
-      return false;
     }
   }
 
-  void EnzymaticDigestion::nextCleavageSite_(const AASequence& protein, AASequence::ConstIterator& iterator) const
+  void EnzymaticDigestion::nextCleavageSite_(const AASequence& protein, AASequence::ConstIterator& iterator, const std::vector<Size> positions) const
   {
     while (iterator != protein.end())
     {
-      if (isCleavageSite_(protein, iterator))
+      if (isCleavageSite_(protein, iterator, positions))
```

Comment:
the code here seems to be inefficient. You already have the positions of the cleavage sites so you should not need to iterate over all positions